qiskit/algorithms/gradients/utils.py

```diff
@@ -0,0 +1,325 @@
+    gradient_circuit_data: ParameterShiftGradientCircuitData,
+) -> List[np.ndarray]:
+    """Makes base parameter values for the parameter shift method. Each base parameter value will
+        be added to the given parameter values in later calculations.
+
+    Args:
+        gradient_circuit_data: gradient circuit data for the base parameter values.
+
+    Returns:
+        The base parameter values for the parameter shift method.
+    """
+    # Make internal parameter values for the parameter shift
+    g_parameters = gradient_circuit_data.gradient_circuit.parameters
+    plus_offsets = []
+    minus_offsets = []
+    # Make base decomposed parameter values for each original parameter
+    for g_param in g_parameters:
+        if g_param in gradient_circuit_data.gradient_virtual_parameter_map:
+            g_param = gradient_circuit_data.gradient_virtual_parameter_map[g_param]
+        idx = g_parameters.data.index(g_param)
+        plus = np.zeros(len(g_parameters))
+        plus[idx] += np.pi / 2
+        minus = np.zeros(len(g_parameters))
+        minus[idx] -= np.pi / 2
+        plus_offsets.append(plus)
+        minus_offsets.append(minus)
+    return plus_offsets + minus_offsets
+
+
+@dataclass
+class LinearCombGradientCircuit:
+    """Gradient circuit for the linear combination of unitaries method.
+
+    Args:
+        gradient_circuit (QuantumCircuit): A gradient circuit  for the linear combination
+            of unitaries method.
+        coeff (float | ParameterExpression): A coefficient corresponds to the gradient circuit.
+    """
+
+    gradient_circuit: QuantumCircuit
+    coeff: float | ParameterExpression
+
+
+def make_lin_comb_gradient_circuit(
+    circuit: QuantumCircuit, add_measurement: bool = False
+) -> Dict[Parameter, List[LinearCombGradientCircuit]]:
+    """Makes gradient circuits for the linear combination of unitaries method.
+
+    Args:
+        circuit: The original quantum circuit.
+        add_measurement: If True, add measurements to the gradient circuit. Defaults to False.
+            ``LinCombSamplerGradient`` calls this method with `add_measurement` is True.
+
+    Returns:
+        A dictionary mapping a parameter to the corresponding list of ``LinearCombGradientCircuit``
+    """
+    supported_gates = [
+        "rx",
+        "ry",
+        "rz",
+        "rzx",
+        "rzz",
+        "ryy",
+        "rxx",
+        "cx",
+        "cy",
+        "cz",
+        "ccx",
+        "swap",
+        "iswap",
+        "h",
+        "t",
+        "s",
+        "sdg",
+        "x",
+        "y",
+        "z",
+    ]
+    circuit2 = transpile(circuit, basis_gates=supported_gates, optimization_level=0)
+
+    qr_aux = QuantumRegister(1, "aux")
+    cr_aux = ClassicalRegister(1, "aux")
+    circuit2.add_register(qr_aux)
+    circuit2.add_bits(cr_aux)
+    circuit2.h(qr_aux)
+    circuit2.data.insert(0, circuit2.data.pop())
+    circuit2.sdg(qr_aux)
+    circuit2.data.insert(1, circuit2.data.pop())
```

Comment:
According to [Julien's suggestion](https://github.com/ikkoham/qiskit-terra/pull/55#discussion_r875979084), you can simplify the code as follows. Could you try it?

```suggestion
    aux = QuantumCircuit(1, 1)
    aux.h(0)
    aux.sdg(0)
    circuit2.tensor(aux, inplace=True)
```

Reply:
It didn't work and I can't figure it what I should fix...